subsys/logging/log_core.c

```diff
@@ -25,6 +26,13 @@ const struct log_backend *uart_backend = &log_backend_uart;
 const struct log_backend *uart_backend;
 #endif
 
+#ifdef CONFIG_LOG_BACKEND_NATIVE_POSIX
+LOG_BACKEND_DEFINE(log_backend_native, log_backend_native_api);
+const struct log_backend *native_backend = &log_backend_native;
+#else
+const struct log_backend *native_backend;
```

Comment:
This may be a bit minor:
Each of this is an unused pointer when its backend is not configured in. I would think a `#define native_backend NULL` would be a better choice.